### PR TITLE
WotLK fixes

### PR DESCRIPTION
**Pull request by ManneN1**
Hello,

I fixed all (?) the bugs related to WotLK.

Noteables:
- The inspect lib is out-of-date and doesn't work for WotLK afaik. I've removed it as a dependency as it was throwing errors.
- I've also updated LibCooldownTracker-1.0 for WotLK, I'll provide a PR for this as well.

Note: I have not tested this on BCC or retail. It might not work there, and the PR might therefore need some more tuning to work correct on the other versions of the game.

### CODE REVIEW

File: Data-Wrath.lua [hunk 1]
```diff
@@ -181,128 +263,325 @@ function GladiusEx.Data.DefaultClassicon()
...
+function GladiusEx.Data.GetSpecializationInfoForClassID(classID, specIndex)
+    local specID = classIDToSpecID[classID][specIndex]
+    local _, name, desc, icon, role, _, _ = GladiusEx.Data.GetSpecializationInfoByID(specID)
+    return specID, name, desc, icon, role, true, true -- isRecommended, isAllowed
+end
+
+function GladiusEx.Data.GetArenaOpponentSpec(id)
+    local unit = "arena" .. id
+
+    if GladiusEx.buttons[unit] and GladiusEx.buttons[unit].specID ~= nil then
```
Comment (vendethiel) [inline Data-Wrath.lua:575]: this is just `return GladiusEx.buttons[unit] and GladiusEx.buttons[unit].specID`

File: GladiusEx.lua [hunk 2]
```diff
@@ -680,33 +683,140 @@ function GladiusEx:PLAYER_ENTERING_WORLD()
...
+                if specID then
+                    return specID
+                end
+            end
+        end
+    end
+end
+
+function GladiusEx:FindSpecBySpell(unit, spellID, spellName)
+    local specID
```
Comment (vendethiel) [inline GladiusEx.lua:763]: this seems to just be
```lua
return GladiusEx.Data.SpecSpells[spellID] or GladiusEx.Data.SpecSpells[spellName]
```

File: modules/cooldowns.lua [hunk 3]
```diff
@@ -14,1139 +14,1295 @@ local GetTime, UnitExists, UnitFactionGroup, UnitClass, UnitRace = GetTime, Unit
...
+            elseif db.trinketIcon == "alliance" and spelldata.icon_alliance then
+                icon = spelldata.icon_alliance
+            elseif db.trinketIcon == "horde" and spelldata.icon_horde then
+                icon = spelldata.icon_horde
+            elseif db.trinketIcon == "allianceWotLk" and spelldata.icon_alliance_wotlk then
+                icon = spelldata.icon_alliance_wotlk
+            elseif db.trinketIcon == "hordeWotLk" and spelldata.icon_horde_wotlk then
+                icon = spelldata.icon_horde_wotlk
+            elseif db.trinketIcon == "factionWotLK" then
+                print(faction)
```
Comment (vendethiel) [inline modules/cooldowns.lua:718]: leftover print statement

File: modules/interrupts.lua [hunk 4]
```diff
@@ -70,19 +70,27 @@ function Interrupt:CombatLogEvent(_, ...)
...
+
+    self:SendMessage("GLADIUSEX_INTERRUPT", unit)
+    
+    if oldTime or duration == nil then return end -- avoid triggering the c_timer again
+    
+    -- K: Clears the interrupt after end of duration (in case no new UNIT_AURA ticks clears it)
+    C_Timer.After(
+        duration + 0.1,
+        function()
+            GladiusEx:GetModule("Interrupts"):UpdateInterrupt(unit, spellid, nil, t+duration+0.1)
```
Comment (vendethiel) [inline modules/interrupts.lua:91]: `GladiusEx:GetModule("Interrupts")` is just `Interrupt`, which is just `self`.

**Comment by vendethiel** [comment]
While I really appreciate this PR, I'm not super fond of the reformatting that was applied...
If I come to merge this, I'd prefer that the formatting was applied everywhere + it has to be something you can call from the CLI (so I can have a pre-commit hook for that style to be coherent in the future).

File: modules/interrupts.lua (hunk 4 above)
Comment (ManneN1) [reply modules/interrupts.lua]: Fixed.

File: modules/cooldowns.lua (hunk 3 above)
Comment (ManneN1) [reply modules/cooldowns.lua]: Fixed.

File: GladiusEx.lua (hunk 2 above)
Comment (ManneN1) [reply GladiusEx.lua]: Fixed.

File: Data-Wrath.lua (hunk 1 above)
Comment (ManneN1) [reply Data-Wrath.lua]: Fixed.

**Comment by ManneN1** [comment]
> While I really appreciate this PR, I'm not super fond of the reformatting that was applied... If I come to merge this, I'd prefer that the formatting was applied everywhere + it has to be something you can call from the CLI (so I can have a pre-commit hook for that style to be coherent in the future).

The formatting in this addon is a mess with or without this PR. It's a mix of tabs, 2-space indent, and 4-space indents everywhere. Additionally line length and other general coding guidelines are also all over the place. This is especially true if you also include the libraries which are used by the addon (thinking mainly of LibCooldownTracker).

Feel free to **apply your own** formatting coherently in the code.

**Comment by vendethiel** [comment]
> The formatting in this addon is a mess with or without this PR. It's a mix of tabs, 2-space indent, and 4-space indents everywhere

And that's why I didn't defend it at all. I didn't disagree with formatting being changed, I disagreed with every contributor having their own formatting.
I'm more than happy to reformat everything, but preferably only once and have it automated for future code changes (that's why I said I prefered something I can run at will).

There's absolutely no point to you applying formatting, then me, then you, except for churn and everything being modified every other PR.

**Comment by ManneN1** [comment]
> There's absolutely no point to you applying formatting, then me, then you, except for churn and everything being modified every other PR. So let's pick one option, stick to it, and things will be fine.

I guess I must've been unclear but I thought I made it clear that I recommend for you to set the formatting of all files to what you prefer in order to avoid the situation as described above.

> Sorry, I think my intention didn't properly come across. The only part I disagree with is formatting being mixed with the rest of changes, because that makes reviewing more difficult, but it's fine with ?w=1 on github.

That's fine, but I needed to fix the formatting of the files I was working with to even be able to work with it without wanting to pull my hair out. I removed most of it before committing.

**Comment by vendethiel** [comment]
> I guess I must've been unclear but I thought I made it clear that I recommend for you to set the formatting of all files to what you prefer in order to avoid the situation as described above.

The only missing thing is to decide on one formatting style. I'm not very knowledgeable about lua formatters, I'm not sure if the formatting you have here is the result of your IDE or of some other program.

**Comment by ManneN1** [comment]
> > I guess I must've been unclear but I thought I made it clear that I recommend for you to set the formatting of all files to what you prefer in order to avoid the situation as described above.
> 
> The only missing thing is to decide on one formatting style. I'm not very knowledgeable about lua formatters, I'm not sure if the formatting you have here is the result of your IDE or of some other program.

I just used https://goonlinetools.com/lua-beautifier/, but https://github.com/marketplace/actions/stylua seems to be a good option if you want it automated (which is naturally preferrable).

**Comment by vendethiel** [comment]
I’ll rerun a format locally.
Thanks for the contribution.

**Comment by vendethiel** [comment]
Hey, did you actually run this on retail at all or not? Seems like it broke cooldowns (at least) because it doesn't know which class/spec people are (even those GLADIUS_SPEC_UPDATE seems to fire just fine).
I have some time again so I need to take a look if you haven't tried it.

**Comment by ManneN1** [comment]
I did not try this on retail as I made very clear in the PR here:

"Note: I have not tested this on BCC or retail. It might not work there, and the PR might therefore need some more tuning to work correct on the other versions of the game."

**Comment by vendethiel** [comment]
You're right, forgot about this when I launched retail again. Pushed retail fixes.

**Pull request merged**